aspnetcore/mvc/advanced/custom-model-binding/sample/CustomModelBindingSample/Binders/AuthorEntityBinder.cs

```diff
@@ -0,0 +1,75 @@
+﻿using CustomModelBindingSample.Data;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CustomModelBindingSample.Binders
+{
+    public class AuthorEntityBinder : IModelBinder
+    {
+        private readonly AppDbContext _db;
+        public AuthorEntityBinder(AppDbContext db)
+        {
+            _db = db;
+        }
+        public Task BindModelAsync(ModelBindingContext bindingContext)
+        {
+            if (bindingContext == null)
+            {
+                throw new ArgumentNullException(nameof(bindingContext));
+            }
+
+            // specify a default argument name if none is set
+            var modelName = bindingContext.ModelName;
+            if(string.IsNullOrEmpty(modelName))
+            {
+                modelName = "authorId";
+            }
```

Comment:
Ok, I didn't know about `bindingContext.BinderModelName`. That helps. I modified things a bit with the last checkin so that the `Author` type specifies the binder type but not the name. Then in the two methods I'm using to demonstrate its usage, I set the name on an action parameter for one (demonstrating this technique) and let it use the default 'authorId' for the other.

Hopefully this looks good now.